components/export/api/src/main/java/org/phenotips/export/internal/DataToCellConverter.java

```diff
@@ -1363,23 +1364,32 @@ public DataSection isSolvedBody(Patient patient)
             return null;
         }
         DataSection bodySection = new DataSection();
-        PatientData<String> patientData = patient.getData("solved");
+        PatientData<SolvedData> patientData = patient.getData("solved");
+        SolvedData data = patientData != null ? patientData.getValue() : null;
 
         int x = 0;
-        if (present.contains("solved")) {
-            String solved = patientData != null ? patientData.get("solved") : null;
+        if (present.contains(SolvedData.STATUS_PROPERTY_NAME)) {
+            String solved = data != null ? data.getStatus() : null;
             DataCell cell = new DataCell(solved != null ? ConversionHelpers.strIntegerToStrBool(solved) : "", x, 0);
             bodySection.addCell(cell);
             x++;
         }
-        if (present.contains("solved__pubmed_id")) {
-            String pubmedId = patientData != null ? patientData.get("solved__pubmed_id") : null;
-            DataCell cell = new DataCell(pubmedId != null ? pubmedId : "", x, 0);
-            bodySection.addCell(cell);
-            x++;
+
+        if (present.contains(SolvedData.PUBMED_ID_PROPERTY_NAME)) {
+            List<String> pubmedIds = data != null ? data.getPubmedIds() : null;
+            if (!pubmedIds.isEmpty()) {
```

Comment:
Won't this give you a NPE if `data` is `null`?

Again, you could use the short syntax on the line before:
`List<String> pubmedIds =  data?.getPubmedIds() ?: Collections.emptyList();`

Reply:
This won't give NPE, both SolvedData class sets pubmedIds to empty list for null inputs.

Reply:
It will give NPE, will fix.

Reply:
for short syntax: same - doesn't compile